apps/src/craft/designer/blocks.js

```diff
@@ -0,0 +1,720 @@
+};
+
+const miniBlocks = Object.keys(miniBlocksToDisplayText).sort();
+
+const soundsToDisplayText = {
+  dig_wood1: i18n.soundTypeDig_wood1(),
+  stepGrass: i18n.soundTypeStepGrass(),
+  stepWood: i18n.soundTypeStepWood(),
+  stepStone: i18n.soundTypeStepStone(),
+  stepGravel: i18n.soundTypeStepGravel(),
+  stepFarmland: i18n.soundTypeStepFarmland(),
+  failure: i18n.soundTypeFailure(),
+  success: i18n.soundTypeSuccess(),
+  fall: i18n.soundTypeFall(),
+  fuse: i18n.soundTypeFuse(),
+  explode: i18n.soundTypeExplode(),
+  placeBlock: i18n.soundTypePlaceBlock(),
+  collectedBlock: i18n.soundTypeCollectedBlock(),
+  bump: i18n.soundTypeBump(),
+  punch: i18n.soundTypePunch(),
+  fizz: i18n.soundTypeFizz(),
+  doorOpen: i18n.soundTypeDoorOpen(),
+  minecart: i18n.soundTypeMinecart(),
+  sheepBaa: i18n.soundTypeSheepBaa(),
+  chickenHurt: i18n.soundTypeChickenHurt(),
+  chickenBawk: i18n.soundTypeChickenBawk(),
+  cowHuff: i18n.soundTypeCowHuff(),
+  cowHurt: i18n.soundTypeCowHurt(),
+  cowMoo: i18n.soundTypeCowMoo(),
+  cowMooLong: i18n.soundTypeCowMooLong(),
+  creeperHiss: i18n.soundTypeCreeperHiss(),
+  ironGolemHit: i18n.soundTypeIronGolemHit(),
+  metalWhack: i18n.soundTypeMetalWhack(),
+  zombieBrains: i18n.soundTypeZombieBrains(),
+  zombieGroan: i18n.soundTypeZombieGroan(),
+  zombieHurt: i18n.soundTypeZombieHurt(),
+};
+
+const allSounds = Object.keys(soundsToDisplayText);
+
+const entityTypesToDisplayText = {
+  Player: i18n.entityTypePlayer(),
+  sheep: i18n.entityTypeSheep(),
+  zombie: i18n.entityTypeZombie(),
+  ironGolem: i18n.entityTypeIronGolem(),
+  creeper: i18n.entityTypeCreeper(),
+  cow: i18n.entityTypeCow(),
+  chicken: i18n.entityTypeChicken(),
+};
+
+const ENTITY_TYPES = Object.keys(entityTypesToDisplayText);
+
+const SPAWNABLE_ENTITY_TYPES = [
+  'sheep',
+  'zombie',
+  'ironGolem',
+  'creeper',
+  'cow',
+  'chicken',
+];
+
+const directionsToDisplay = {
+  up: i18n.directionUp(),
+  middle: i18n.directionMiddle(),
+  down: i18n.directionDown(),
+  left: i18n.directionLeft(),
+  right: i18n.directionRight(),
+};
+
+function keysToDropdownOptions(keysList) {
+  return keysList.map(function (key) {
+    var displayText = (blocksToDisplayText[key] ||
+      numbersToDisplayText[key] ||
+      directionsToDisplay[key] ||
+      entityTypesToDisplayText[key] ||
+      miniBlocksToDisplayText[key] ||
+      key);
+    return [displayText, key];
+  });
+}
+
+
+const entityActionBlocks = {
+  destroyEntity: i18n.blockActionDestroyEntity(),
+  attack: i18n.blockActionAttack(),
+  flashEntity: i18n.blockActionFlashEntity(),
+  moveForward: i18n.blockActionMoveForward(),
+  moveRandom: i18n.blockActionMoveRandom(),
+  explodeEntity: i18n.blockActionExplodeEntity()
+};
+
+const entityActionTargetDropdownBlocks = {
+  'moveToward': i18n.blockActionMoveToward(),
+  'moveTo': i18n.blockActionMoveTo(),
+  'moveAway': i18n.blockActionMoveAway(),
+};
+
+exports.entityActionBlocks = Object.keys(entityActionBlocks);
```

Comment:
will stop commenting on ES6 usage, but it seems like we're using it inconsistently, or perhaps just using a different subset of it.